OpenEdXMobile/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -133,24 +133,29 @@ private void init() {
     private void checkIfAppVersionUpgraded(Context context) {
         PrefManager.AppInfoPrefManager prefManager = new PrefManager.AppInfoPrefManager(context);
         long previousVersionCode = prefManager.getAppVersionCode();
-        final long currVersionCode = (long) BuildConfig.VERSION_CODE;
-        if (previousVersionCode < currVersionCode) {
+        final long curVersionCode = (long) BuildConfig.VERSION_CODE;
+        if (previousVersionCode < 0) {
```

Comment:
This logic is wrong with respect to what we had before. Previously we called the `injector.getInstance(IStorage.class).repairDownloadCompletionData()` function whenever `previousVersionCode < curVersionCode` was true.

Due to this new check `repairDownloadCompletionData` won't get called on atleast 3 occasions where it was previously called:

1. On app's fresh install.
2. When a user logs out and reopens the app (p.s. we clear all user preferences on logout).
3. If user had a really old version of app and updated directly to the version of app in which this change got implemented.